SPEC.md

```diff
@@ -0,0 +1,195 @@
+    * The `kind` and `kindShort` labels have two segments: a prefix and a name, separated by a slash (/).
+    * The name segment is required and must be 63 characters or less, beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (\_), dots (.), and alphanumerics between.
+    * The prefix must be a DNS subdomain: a series of DNS labels separated by dots (.), not longer than 253 characters in total, followed by a slash (/).
+    * Examples (not an exhaustive list):
+      * Valid: `vendor.com/foo`, `foo.bar.baz/foo-bar123.B_az`.
+      * Invalid: `foo`, `vendor.com/foo/`, `vendor.com/foo/bar`.
+
+#### Container Runtime
+
+* `containerRuntime` (array of string, OPTIONAL) indicates that this CDI specification targets only a specific container runtime.
+  If this field is not indicated then container runtimes should consider that the JSON file targets all runtimes.
+  If this field is indicated then container runtimes should only run it if one of the identifiers matches the container runtime's identifier.
+  Possible values (not an exhaustive list): docker, podman, gvisor, lxc
+
+#### CDI Devices
+
+The `cdiDevices` field describes the set of hardware devices that can be requested by the container runtime user.
+Note: For a CDI file to be valid, at least one entry must be specified in this array.
+
+  * `cdiDevices` (array of objects, REQUIRED) list of devices provided by the vendor.
+    * `name` (string, REQUIRED), name of the device, can be used to refer to it when requesting a device.
+      * Beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (\_), dots (.), and alphanumerics between.
+      * e.g: `docker/podman run --device foo ...`
+    * `nameShort` (array of strings, OPTIONAL), alternative names for the device. Can be used to reduce the CLI verbosity
+      * Entries in the array MUST use the same schema as the entry for the `name` field
+    * `containerSpec` (object, OPTIONAL) this field is described in the next section.
+      * This field should only be merged in the OCI spec if the device has been requested by the container runtime user.
+
+
+#### Container Spec
+
+The `containerSpec` field describes edits to be made to the OCI specification. Currently only three kinds of edits can be made to the OCI specification: `devices`, `mounts` and `hooks`.
+
+The `containerSpec` field is referenced in two places in the specification:
+  * At the device level, where the edits MUST only be made if the matching device is requested by the container runtime user.
+  * At the container level, where the edits MUST be made if any of the of the device defined in the `devices` field are requested.
+
+
+The `containerSpec` field has the following definition:
+  * `devices` (array of objects, OPTIONAL) describes the device nodes that should be mounted:
+    * `hostPath` (string, REQUIRED) path of the device on the host.
```

Comment:
`hostPath` implies that "all" needed files are available on the host. Should we consider other sources, like images or running containers? If a running container provides all needed files why go the detour over the host. This can complicate things, especially with SELinux and AppArmor.

Reply:
I think that's definitely a question we need to address, let me open an issue for that!

Reply:
Opened: https://github.com/container-orchestrated-devices/container-device-interface/issues/3